ext/phar/phar.c

```diff
@@ -1112,19 +1192,15 @@ static int phar_parse_pharfile(php_stream *fp, char *fname, size_t fname_len, ch
 		}
 
 		PHAR_GET_32(buffer, len);
-		if (entry.is_persistent) {
-			entry.metadata_len = len;
-		} else {
-			entry.metadata_len = 0;
-		}
 		if (len > (size_t)(endbuffer - buffer)) {
 			pefree(entry.filename, entry.is_persistent);
 			MAPPHAR_FAIL("internal corruption of phar \"%s\" (truncated manifest entry)");
 		}
-		if (phar_parse_metadata(&buffer, &entry.metadata, len) == FAILURE) {
-			pefree(entry.filename, entry.is_persistent);
-			MAPPHAR_FAIL("unable to read file metadata in .phar file \"%s\"");
-		}
+		/* Don't implicitly call unserialize() on potentially untrusted input unless getMetadata() is called directly. */
+		/* The same local variable entry is reused in a loop, so reset the state before reading data. */
+		ZVAL_UNDEF(&entry.metadata_tracker.val);
+		entry.metadata_tracker.str = NULL;
```

Comment:
Can this go through phar_metadata_tracker_free ... somewhere?

Reply:
> 		`/* The same local variable entry is reused in a loop, so reset the state before reading data. */`

I could call memset to set all of the bytes of `entry` to 0 in this loop, and do the initialization inside of the loop body instead of initializing outside and resetting *just* the fields that changed within the loop. Was waiting to see what other people's thoughts were - doubt the performance impact would be noticeable of the alternative I suggest.

Reply:
I think your current code is also fine.